polyply/src/backmap.py

```diff
@@ -11,15 +11,128 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
+import numpy as np
+import scipy.optimize
+import networkx as nx
 from .processor import Processor
 from .generate_templates import find_atoms
+from .linalg_functions import rotate_xyz
 """
 Processor implementing a template based back
 mapping to lower resolution coordinates for
 a meta molecule.
 """
 
+def find_edges(molecule, attr, value):
+    """
+    Find all edges of a `vermouth.molecule.Molecule` that have the
+    attribute `attr` with the corresponding value of `value`. Note
+    that this function is not order specific. That means (1, 2) is
+    the same as (2, 1).
+
+    Parameters
+    ----------
+    molecule: :class:vermouth.molecule.Molecule
+    attrs: tuple(str, str)
+         tuple of the attributes used in matching
+    values: tuple
+         corresponding values value
+
+    Returns
+    ----------
+    list
+       list of edges found
+    """
+    edges = []
+    for edge in molecule.edges:
+        node_a, node_b = edge
+
+        if attr[0] in molecule.nodes[node_a]:
+           if molecule.nodes[node_a][attr[0]] == value[0]:
+              if  attr[1] in molecule.nodes[node_b]:
+                 if molecule.nodes[node_b][attr[1]] == value[1]:
+                     edges.append(edge)
+
+        if attr[1] in molecule.nodes[node_a]:
+           if molecule.nodes[node_a][attr[1]] == value[1]:
+               if attr[0] in molecule.nodes[node_b]:
+                  if molecule.nodes[node_b][attr[0]] == value[0]:
+                      edges.append(edge)
```

Comment:
This makes my eyes bleed. And you'll possibly end up with *both* (1, 2) and (2, 1).

```suggestion
        nodes = [molecule.nodes[node_a], molecule.nodes[node_b]]
        if all(node[at] == val for (node, at, val) in zip(nodes, attr, value)) or all(node[at] == val for (node, at, val) in zip(reversed(nodes), attr, value)):
            edges.append(edge)
```

You can still replace the `or` by a call to `any`. And you may want to add a line break somewhere